include/gsl/span

```diff
@@ -253,27 +258,27 @@ namespace details
             return !(rhs > lhs);
         }
 
+#ifdef _MSC_VER
+        // Tell Microsoft standard library how to unwrap span iterators
```

Comment:
@annagrin, given our discussion of the impedance mismatch - `span` always wants range checking, STLSTL only range checks when compiling `_DEBUG` - would you prefer that I make this part of the change conditional on `_DEBUG`, or just rip it out for now?

Reply:
To be clear: the rest of this change doesn't interfere with `span`'s range-checking behavior, assuming that you trust the compiler no to miscompile range-based for loops.

Reply:
@CaseyCarter Let's wait with the range checking until the impedance mismatch is resolved. If you would like to submit just the range-based loops change, that would be great! Thanks, the code gen looks awesome with the change.